src/herder/TxSetFrame.cpp

```diff
@@ -572,7 +573,27 @@ applySurgePricing(TxSetPhase phase, TxFrameList const& txs, Application& app
             TxStageFrameList frameList;
             if (!txs.empty())
             {
-                frameList = {{txs}};
+                for (auto const& stageIndexes : parallelSorobanOrder)
+                {
+                    TxStageFrame stage;
+                    for (auto const& threadIndexes : stageIndexes)
+                    {
+                        TxFrameList threadTxs;
+                        for (auto const& txIndex : threadIndexes)
+                        {
+                            threadTxs.push_back(txs.at(txIndex));
```

Comment:
I don't want to block this PR, but I think it's going to be easier to set up more complex tests if we passed a single nested vector with tx frames instead for the indices (as a separate function or overload)

Reply:
FWIW I mean not duplication, but passing a single nested vector of tx frames.